ddf-test/src/main/scala/io/ddf/test/it/analytics/AggregationHandlerBaseSuite.scala

```diff
@@ -47,26 +60,151 @@ trait AggregationHandlerBaseSuite extends BaseSuite with Matchers {
   }
 
-  test("group and aggregate)2 steps") {
-    val ddf = loadAirlineDDF()
+  test("group and aggregate 2 steps") {
+    val ddf = loadAirlineDDF(useCache = true)
     val ddf2 = ddf.getAggregationHandler.groupBy(List("DayofMonth"))
     val result = ddf2.getAggregationHandler.agg(List("mean=avg(ArrDelay)"))
     result.getColumnNames.map(col => col.toLowerCase) should (contain("mean") and contain("dayofmonth"))
     val rows = result.sql("select * from @this", "").getRows
     rows.head.split("\t").head.toDouble should be(9.0 +- 1.0)
   }
 
-  test("throw an error on aggregate without groups") {
-    val airlineDDF = loadAirlineDDF()
-    val ddf = manager.sql2ddf("select * from airline", engineName)
-    intercept[Exception] {
-      ddf.getAggregationHandler.agg(List("mean=avg(ArrDelay)"))
+  test("calculate correlation") {
+    val ddf = loadAirlineDDF(useCache = true)
+    ddf.correlation("ArrDelay", "DepDelay") should be(0.89 +- 1)
+  }
+
+  test("Proper error message for non-existent columns") {
+    val ddf = loadAirlineDDF(useCache = true)
+
+    val thrown = intercept[DDFException] {
+      ddf.groupBy(List("Year1"), List("count(*)"))
+    }
+    assert(thrown.getMessage === "Non-existent column: Year1")
+
+    val thrown2 = intercept[DDFException] {
+      ddf.groupBy(List("Year"), List("avg(arrdelay1)"))
     }
+    assert(thrown2.getMessage === "Non-existent column: arrdelay1")
+
+    val thrown3 = intercept[DDFException] {
+      ddf.aggregate("count(*),Year1")
+    }
+    assert(thrown3.getMessage === "Non-existent column: Year1")
+
+    val thrown4 = intercept[DDFException] {
+      ddf.aggregate("avg(arrdelay1),Year")
+    }
+    assert(thrown4.getMessage === "Non-existent column: arrdelay1")
   }
 
-  test("calculate correlation") {
-    val ddf = loadAirlineDDF()
-    ddf.correlation("ArrDelay", "DepDelay") should be(0.89 +- 1)
+  test("Proper error message for expressions") {
+    val ddf = loadAirlineDDF(useCache = true)
+
+    val thrown1 = intercept[DDFException] {
+      ddf.groupBy(List("Year"), List("arrdelay+"))
+    }
+    assert(thrown1.getMessage === "Column or Expression with invalid syntax: 'arrdelay+'")
+
+    val thrown2 = intercept[DDFException] {
+      ddf.aggregate("arrdelay+,Year")
+    }
+    assert(thrown2.getMessage === "Column or Expression with invalid syntax: 'arrdelay+'")
+  }
+
+  test("Proper error message for undefined function") {
+    val ddf = loadAirlineDDF(useCache = true)
+
+    val thrown1 = intercept[DDFException] {
+      ddf.groupBy(List("Year"), List("aaa(arrdelay)"))
+    }
+    assert(thrown1.getMessage === "undefined function aaa")
+  }
+
+  test("Proper error message for wrong Hive UDF usage") {
+    val ddf = loadAirlineDDF(useCache = true)
+
+    val thrown1 = intercept[DDFException] {
+      ddf.groupBy(List("Year"), List("substring('aaaa')"))
+    }
+    assert(thrown1.getMessage.contains("No matching method for class org.apache.hadoop.hive.ql.udf.UDFSubstr with " +
```

Comment:
This error message is specific for DDF Spark and should not be in the Base Test, @hai-adatao.